mvc_controller.cpp

```diff
@@ -331,20 +331,33 @@ void MvcController::ConditionallyEnableItems
         }
     else if(itembox)
         {
-        // WX !! wxWindowUpdateLocker doesn't seem to help much.
-        wxWindowUpdateLocker u(&control);
-        itembox->Clear();
-        // WX !! Append(wxArrayString const&) "may be much faster"
-        // according to wx online help, but that seems untrue: its
-        // implementation just uses a loop.
+        auto& cached_items = itemboxes_cache_[name];
+
+        std::vector<wxString> items;
+        items.reserve(datum->cardinality());
+
         for(int j = 0; j < datum->cardinality(); ++j)
             {
             if(datum->is_allowed(j))
                 {
-                itembox->Append(datum->str(j));
+                items.push_back(datum->str(j));
                 }
             }
-        itembox->SetStringSelection(datum->str(datum->ordinal()));
+
+        wxWindowUpdateLocker u(&control);
+
+        if(items != cached_items)
+            {
+            std::swap(cached_items, items);
+
+            itembox->Set(cached_items);
+            }
+
+        auto selected_string = datum->str(datum->ordinal());
```

Comment:
```suggestion
        auto const& selected_string = datum->str(datum->ordinal());
```
Micro optimization to avoid a string copy.

Reply:
`datum->str` return new string, not const reference:
```cpp
class LMI_SO mc_enum_base
    :public datum_base
{
  public:
...
    virtual std::string str(int) const = 0;
```

Reply:
Yes, binding a const reference to it avoids copying the temporary object (please read [GotW 88](https://herbsutter.com/2008/01/01/gotw-88-a-candidate-for-the-most-important-const/) if you have any doubts about how this works).

Reply:
Ok, thanks for the link.